renovate.json

```diff
@@ -1,6 +1,89 @@
 {
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
-  "extends": [
-    "config:recommended"
-  ]
+  "labels": [
+    "renovatebot",
+    "dependencies"
+  ],
+  "constraints": {
+    "go": "1.22"
+  },
+  "schedule": ["every tuesday"],
+  "extends": ["config:recommended"],
+  "packageRules": [
+    {
+      "matchManagers": ["gomod"],
+      "matchUpdateTypes": ["pin", "pinDigest", "digest", "lockFileMaintenance", "rollback", "bump", "replacement"],
+      "enabled": false
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchUpdateTypes": ["major"],
+      "prBodyNotes": [":warning: MAJOR VERSION UPDATE :warning: - please manually update this package"],
+      "labels": ["dependency-major-update"]
+    },
+    {
+      "matchManagers": ["dockerfile"],
+      "groupName": "dockerfile deps"
+    },
+    {
+      "matchManagers": ["docker-compose"],
+      "groupName": "docker-compose deps"
+    },
+    {
+      "matchManagers": ["github-actions"],
+      "groupName": "github-actions deps"
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchSourceUrls": [
+        "https://github.com/open-telemetry/opentelemetry-go-contrib"
+      ],
+      "groupName": "All opentelemetry-go-contrib packages"
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchSourceUrlPrefixes": ["https://go.opentelemetry.io/otel"],
+      "groupName": "All go.opentelemetry.io/contrib packages"
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchSourceUrlPrefixes": ["https://google.golang.org"],
+      "groupName": "All google.golang.org packages"
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchPackagePrefixes": ["golang.org/x"],
+      "groupName": "All golang.org/x packages"
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchPackagePrefixes": ["go.opentelemetry.io/collector"],
+      "groupName": "All go.opentelemetry.io/collector packages"
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchPackagePrefixes": ["go.opentelemetry.io/build-tools"],
+      "groupName": "All go.opentelemetry.io/build-tools packages"
+    },
+    {
+      "matchManagers": ["gomod"],
+      "matchDepTypes": ["toolchain"],
+      "enabled": false
+    }
+  ],
+  "customManagers": [
+    {
+      "customType": "regex",
+      "fileMatch": [
+        "(^|\\/).github\\/.*\\.ya?ml$"
+      ],
+      "matchStrings": [
+        "goreleaser\\/goreleaser-action@[\\S\\s]+version: (?<currentValue>.*?)$"
+      ],
+      "depNameTemplate": "github.com/goreleaser/goreleaser-pro",
+      "datasourceTemplate": "github-releases"
+    }
```

Comment:
This is the part that's new to update the goreleaser-pro version in the github action.
The rest of the PR is the copied renovate config from collector-core repo.